service/src/main/java/greencity/service/FriendServiceImpl.java

```diff
@@ -1,15 +1,49 @@
 package greencity.service;
 
+import greencity.constant.ErrorMessage;
+import greencity.dto.PageableDto;
+import greencity.dto.user.UserFriendDto;
+import greencity.entity.User;
+import greencity.exception.exceptions.NotFoundException;
 import greencity.repository.UserRepo;
-import javax.transaction.Transactional;
+
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
 
 @Slf4j
 @Service
-@Transactional
+@Transactional(readOnly = true)
 @RequiredArgsConstructor
 public class FriendServiceImpl implements FriendService {
     private final UserRepo userRepo;
+    private final ModelMapper modelMapper;
+
+    public PageableDto<UserFriendDto> findAllUsersFriends(Long userId, Pageable pageable) {
+
+        List<User> friends = userRepo.getAllUserFriends(userId);
+        if (friends.isEmpty()) {
```

Comment:
I think you need to throw an exception when a user passes a non-existing userId and just send an empty PageableDto if the user doesn`t have friends.

Reply:
I use current user in the controller that why a current user can't be non-existent. For front-end work like expected task need response 404 in case a user doesn't have friends

Reply:
Front-end will see that list is empty and not found "The HTTP 404 Not Found response status code indicates that the server cannot find the requested resource." so I would better check if userId exist in db